connector-sdk/document/src/main/java/io/camunda/operation/IntrinsicOperationExecutor.java

```diff
@@ -14,8 +14,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.connector.document.annotation.jackson;
+package io.camunda.operation;
 
-import java.util.function.Supplier;
+public interface IntrinsicOperationExecutor {
 
-public interface DocumentOperationResult<T> extends Supplier<T> {}
+  <T> T execute(String operationName, IntrinsicOperationParams params, Class<T> resultType);
+}
```

Comment:
The executor receives the `resultType`? Shouldnt that come from the operation itself?

Reply:
I thought of this as a way for the caller to say "I want the result of the function to be of this type, otherwise throw an exception", which is a common use case in our deserializers. But I'm with you here, we can remove this, and also I think the result type conversion via Jackson is problematic from the security perspective (typical injection attack if we use `parseJson` and the document contains a JSON which defines an operation)